astropy/modeling/bounding_box.py

```diff
@@ -17,12 +17,16 @@
 from astropy.utils import isiterable
 
 if TYPE_CHECKING:
+    import sys
     from typing import Any, Callable
 
-    from typing_extensions import Self
-
     from astropy.units import UnitBase
 
+    if sys.version_info >= (3, 11):
+        from typing import Self
+    else:
+        from typing_extensions import Self
```

Comment:
This `if TYPE_CHECKING:` block was added in #15354. An early version of that pull request did include explicitly version dependent code, but during the review @nstarman pointed out that Ruff (specifically [UP035](https://docs.astral.sh/ruff/rules/deprecated-import/)) knows enough about `typing` and `typing_extensions` that it can automatically change the import when support for Python 3.10 is dropped. The change proposed here is not necessary.

Reply:
#minimum-effort.

Reply:
Is this conversation resolved or not? Is this PR good to merge? @eerovaher ?